osh/builtin_process2.py

```diff
@@ -57,6 +61,133 @@ def Run(self, cmd_val):
+      Returns the status of the last ID; fails if ID is invalid or an invalid
+      option is given.
+  """
+  def __init__(self, waiter, job_state, mem, tracer, errfmt):
+    # type: (Waiter, JobState, Mem, dev.Tracer, ErrorFormatter) -> None
+    self.waiter = waiter
+    self.job_state = job_state
+    self.mem = mem
+    self.tracer = tracer
+    self.errfmt = errfmt
+
+  def Run(self, cmd_val):
+    # type: (cmd_value__Argv) -> int
+    with dev.ctx_Tracer(self.tracer, 'wait', cmd_val.argv):
+      return self._Run(cmd_val)
+
+  def _Run(self, cmd_val):
+    # type: (cmd_value__Argv) -> int
+    attrs, arg_r = flag_spec.ParseCmdVal('wait', cmd_val)
+    arg = arg_types.wait(attrs.attrs)
+
+    job_ids, arg_spids = arg_r.Rest2()
+
+    if arg.n:
+      # Loop until there is one fewer process running, there's nothing to wait
+      # for, or there's a signal
+      n = self.job_state.NumRunning()
+      if n == 0:
+        status = 127
+      else:
+        target = n - 1
+        status = 0
+        while self.job_state.NumRunning() > target:
+          result = self.waiter.WaitForOne()
+          if result == process.W1_OK:
+            status = self.waiter.last_status
+          elif result == process.W1_ECHILD:
+            # nothing to wait for, or interrupted
+            status = 127
+            break
+          elif result >= 0:  # signal
+            status = 128 + result
+            break
+
+      return status
+
+    if len(job_ids) == 0:
+      #log('*** wait')
+
+      # BUG: If there is a STOPPED process, this will hang forever, because we
+      # don't get ECHILD.  Not sure it matters since you can now Ctrl-C it.
+      # But how to fix this?
+
+      status = 0
+      while self.job_state.NumRunning() != 0:
+        result = self.waiter.WaitForOne()
+        if result == process.W1_ECHILD:
+          # nothing to wait for, or interrupted.  status is 0
+          break
+        elif result >= 0:  # signal
+          status = 128 + result
+          break
+
+      return status
+
+    # Get list of jobs.  Then we need to check if they are ALL stopped.
+    # Returns the exit code of the last one on the COMMAND LINE, not the exit
+    # code of last one to FINISH.
+    status = 1  # error
+    for i, job_id in enumerate(job_ids):
+      span_id = arg_spids[i]
+
+      # The % syntax is sort of like ! history sub syntax, with various queries.
+      # https://stackoverflow.com/questions/35026395/bash-what-is-a-jobspec
+      if job_id.startswith('%'):
+        raise error.Usage(
+            "doesn't support bash-style jobspecs (got %r)" % job_id,
+            span_id=span_id)
+
+      # Does it look like a PID?
+      try:
+        pid = int(job_id)
+      except ValueError:
+        raise error.Usage('expected PID or jobspec, got %r' % job_id,
+                          span_id=span_id)
+
+      job = self.job_state.JobFromPid(pid)
+      if job is None:
+        self.errfmt.Print_("%d isn't a child of this shell" % pid,
```

Comment:
This `%d` used to be `%s`. `pid` is an int so there was a compiler error in the generated code